typeid/typeid-go/encoding.go

```diff
@@ -25,7 +25,7 @@ func (tid *TypeID[P]) UnmarshalText(text []byte) error {
 
 // MarshalText implements the encoding.TextMarshaler interface.
 // It encodes a TypeID as a string using the same logic as String()
-func (tid TypeID[P]) MarshalText() (text []byte, err error) {
+func (tid TypeID) MarshalText() (text []byte, err error) {
```

Comment:
Implement `encoding.TextAppender` and make this `return tid.AppendText(nil)`?

Reply:
Good idea. Doing.